File: GithubUserJava/app/src/main/java/com/example/myapplication/factory/DetailViewModelFactory.java
package com.example.myapplication.factory;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.example.myapplication.viewmodel.DetailViewModel;

public class DetailViewModelFactory implements ViewModelProvider.Factory {

    private final String username;
    private final Application app;

    public DetailViewModelFactory(String username, Application app) {
        this.username = username;
        this.app = app;
    }

    @NonNull
    @Override
    @SuppressWarnings("unchecked")
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(DetailViewModel.class)) {
            return (T) new DetailViewModel(username, app);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: GithubUserKotlin/app/src/main/java/com/example/myapplication/factory/DetailViewModelFactory.kt
package com.example.myapplication.factory

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.myapplication.viewmodel.DetailViewModel

class DetailViewModelFactory(private val username: String, private val app: Application) : ViewModelProvider.Factory {
    @Suppress("UNCHECKED_CAST")

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(DetailViewModel::class.java)) {
            return DetailViewModel(username,app) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }

}